website/docs/r/compute_disk.html.markdown

```diff
@@ -1,102 +1,231 @@
-
-* `zone` - (Required) The zone where this disk will be available.
-
-- - -
+Add a persistent disk to your instance when you need reliable and
+affordable storage with consistent performance characteristics.
 
-* `disk_encryption_key_raw` - (Optional) A 256-bit [customer-supplied encryption key]
-    (https://cloud.google.com/compute/docs/disks/customer-supplied-encryption),
-    encoded in [RFC 4648 base64](https://tools.ietf.org/html/rfc4648#section-4)
-    to encrypt this disk.
 
-* `image` - (Optional) The image from which to initialize this disk. This can be
-    one of: the image's `self_link`, `projects/{project}/global/images/{image}`,
-    `projects/{project}/global/images/family/{family}`, `global/images/{image}`,
-    `global/images/family/{family}`, `family/{family}`, `{project}/{family}`,
-    `{project}/{image}`, `{family}`, or `{image}`. If referred by family, the
-    images names must include the family name. If they don't, use the
-    [google_compute_image data source](/docs/providers/google/d/datasource_compute_image.html).
-    For instance, the image `centos-6-v20180104` includes its family name `centos-6`.
-    These images can be referred by family name here.
 
-* `project` - (Optional) The ID of the project in which the resource belongs. If it
-    is not provided, the provider project is used.
+## Argument Reference
 
-* `size` - (Optional) The size of the image in gigabytes. If not specified, it
-    will inherit the size of its base image.
+The following arguments are supported:
 
-* `snapshot` - (Optional) Name of snapshot from which to initialize this disk.
+* `zone` -
+  (Required)
+  A reference to the zone where the disk resides.
 
-* `type` - (Optional) The GCE disk type.
 
-* `labels` - (Optional) A set of key/value label pairs to assign to the image.
+- - -
 
+* `description` -
+  (Optional)
+  An optional description of this resource. Provide this property when
+  you create the resource.
+* `labels` -
+  (Optional)
+  Labels to apply to this disk.  A list of key->value pairs.
+* `name` -
+  (Optional)
+  Name of the resource. Provided by the client when the resource is
+  created. The name must be 1-63 characters long, and comply with
+  RFC1035. Specifically, the name must be 1-63 characters long and match
+  the regular expression `[a-z]([-a-z0-9]*[a-z0-9])?` which means the
+  first character must be a lowercase letter, and all following
+  characters must be a dash, lowercase letter, or digit, except the last
+  character, which cannot be a dash.
+* `size` -
+  (Optional)
+  Size of the persistent disk, specified in GB. You can specify this
+  field when creating a persistent disk using the sourceImage or
+  sourceSnapshot parameter, or specify it alone to create an empty
+  persistent disk.
+
+  If you specify this field along with sourceImage or sourceSnapshot,
+  the value of sizeGb must not be less than the size of the sourceImage
+  or the size of the snapshot.
+* `image` -
```

Comment:
We have a bunch of ways we allow specifying the image, this may need an override so we can put them in

Reply:
Done.